### PR TITLE
Feature/csl700 sensitive logging

### PR DESCRIPTION
**Pull request by volhovM**

### CODE REVIEW

File: src/Pos/Util/LogSafe.hs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    dispatchMessage
+        (loggerName      -> name)
+        (convertSeverity -> prior)
+        msg =
+      let acceptable p
+              | "fileHandler" `isPrefixOf` p ||
+                "rollerHandler" `isPrefixOf` p
+              = not $ ".pub" `isSuffixOf` p
+              | otherwise = True
+      in liftIO $ L.logMCond name prior (T.unpack msg) acceptable
```
Comment (gromakovsky) [inline src/Pos/Util/LogSafe.hs]: Why do you use `T.unpack` instead of `toString`? 😕

File: src/Pos/Util/LogSafe.hs [hunk 2]
```diff
@@ -7,37 +7,47 @@ module Pos.Util.LogSafe
...
 execSecureLogWrapped :: SecureLogWrapped m a -> m a
 execSecureLogWrapped (SecureLogWrapped act) = act
 
-class HasSecureFormatting a where
-    secureF :: Bool -> Format r (a -> r)
-
 -- | Shortcut for 'logMessage' to use according severity.
 logDebugS, logInfoS, logNoticeS, logWarningS, logErrorS
-    :: WithLogger m
+    :: (HasLoggerName m, MonadIO m)
```
Comment (gromakovsky) [inline src/Pos/Util/LogSafe.hs]: Are we sure we won't need a secret pure logging?

Comment (volhovM) [inline src/Pos/Util/LogSafe.hs]: Most definitely. I barely managed to find a solution to make this happen (by forking `hslogger`). I don't know if it's easy to support secret logging in pure environment.

Comment (gromakovsky) [inline src/Pos/Util/LogSafe.hs]: What's the problem? Is it because we need a filter (like `acceptable`)?

**Comment by georgeee** [comment]
@volhovM where are actual changes of codebase (i.e. usages of these new secret logging functions)?

**Comment by volhovM** [comment]
@georgeee none, i suppose this should be done as part of CSL-837.

**Comment by gromakovsky** [comment]
@volhovM then change tittle of CSL-837 and also I am not sure 1 day will be enough.

File: src/Pos/Util/LogSafe.hs (hunk 2 above)
Comment (gromakovsky) [inline src/Pos/Util/LogSafe.hs]: @volhovM so?

**Comment by georgeee** [comment]
Updated CSL-837 description

**Pull request merged**